Add get_uploaded_entity helper

ttung opened the pull request:
### Description
This simplifies the logic when we're going from sample to whatever was uploaded to represent the sample.

Depends on #230 

### Test plan
make -j style unit-tests

phoenixAja commented on src/py/aspen/database/models/sample.py:
```diff
@@ -167,3 +167,14 @@ class Sample(idbase, DictMixin):
+            return self.sequencing_reads_collection
+        elif self.uploaded_pathogen_genome is not None:
+            return self.uploaded_pathogen_genome
+        raise ValueError(
+            "Sample has neither sequencing reads nor uploaded pathogen genome."
+        )
```
this is much nicer!!

phoenixAja commented:
this is much nicer! Thanks for adding!

Commit:
Add get_uploaded_entity helper

### Description
This simplifies the logic when we're going from sample to whatever was uploaded to represent the sample.

### Test plan
make -j style unit-tests
